modules/account-lib/src/coin/sol/utils.ts

```diff
@@ -204,26 +231,14 @@ export function requiresAllSignatures(signatures: SignaturePubkeyPair[]): boolea
 export function getTransactionType(transaction: SolTransaction): TransactionType {
   const { instructions } = transaction;
   validateIntructionTypes(instructions);
-  if (
-    instructions.length <= 3 &&
-    getInstructionType(instructions[walletInitInstructionIndexes.Create]) === ValidInstructionTypesEnum.Create &&
-    getInstructionType(instructions[walletInitInstructionIndexes.InitializeNonce]) ===
-      ValidInstructionTypesEnum.InitializeNonceAccount
-  ) {
+  if (matchTransactionTypeByInstructionsOrder(instructions, walletInitInstructionIndexes)) {
     return TransactionType.WalletInitialization;
-  } else if (
-    getInstructionType(instructions[stakingActivateInstructionsIndexes.Create]) === ValidInstructionTypesEnum.Create &&
-    getInstructionType(instructions[stakingActivateInstructionsIndexes.Initialize]) ===
-      ValidInstructionTypesEnum.StakingInitialize &&
-    getInstructionType(instructions[stakingActivateInstructionsIndexes.Delegate]) ===
-      ValidInstructionTypesEnum.StakingDelegate
-  ) {
+  } else if (matchTransactionTypeByInstructionsOrder(instructions, stakingActivateInstructionsIndexes)) {
```

Comment:
As `if` content ends in a return, `else` is not required. The same logic applies to l238 and l240

Reply:
nit, I take it into account since I probably change this fn after talk with Rodrigo